app/services/clearancing_service.rb

```diff
@@ -0,0 +1,47 @@
+require 'csv'
+require 'ostruct'
+class ClearancingService
+
+  attr_accessor :clearancing_status
+
+  def initialize
+    self.clearancing_status = create_clearancing_status
+  end
+
+  def process_items(potential_item_ids)
+    potential_item_ids.each do |potential_item_id|
+      item_id_is_valid, error_message = clearancing_status.clearance_batch.item_id_is_valid?(potential_item_id.to_i)
```

Comment:
Since you're performing the `item_id_is_valid?` check when you loop through the items on line 29, I'm not sure you need to go through them all here. You could remove this check and collect the error messages when you perform the check on line 29, which would save you from going through the list twice (and might remove the need for two separate methods). If you'd prefer to keep the validity check separate from the clearancing itself, though, you could just remove the redundant check from line 29.